src/PluginForm/Bluesnap/HostedPaymentFieldsPaymentMethodAddForm.php

```diff
@@ -0,0 +1,214 @@
+      $entity_type_manager,
+      $logger
+    );
+
+    $this->apiService = $api_service;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.commerce_inline_form'),
+      $container->get('current_route_match'),
+      $container->get('entity_type.manager'),
+      $container->get('logger.factory')->get('commerce_payment'),
+      $container->get('commerce_bluesnap.api_service')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildCreditCardForm(
+    array $element,
+    FormStateInterface $form_state
+  ) {
+    // The BlueSnap unique Hosted Payment Fields Token that will be sent in the
+    // drupalSettings to the JS.
+    // First, initialize BlueSnap.
+    if (empty($form_state->getValue('bluesnap_token'))) {
+      $plugin = $this->entity->getPaymentGateway()->getPlugin();
+      $bluesnap_token = $this->apiService->getHostedPaymentFieldsToken($plugin);
+    }
+    else {
+      $bluesnap_token = $form_state->getValue('bluesnap_token');
+    }
+
+    // Alter the form with Bluesnap specific needs.
+    $element['#attributes']['class'][] = 'bluesnap-form';
+    $element['#attached']['library'][] = 'commerce_bluesnap/hosted_payment_fields_form';
+    $element['#attached']['drupalSettings']['commerceBluesnap'] = [
+      'hostedPaymentFields' => [
+        'token' => $bluesnap_token,
+      ],
+    ];
+
+    // Hidden fields which will be populated by the js.
+    $this->hiddenFields($element);
+    // The credit card fields necessary for BlueSnap.
+    $this->ccFields($element);
+
+    // To display validation errors.
+    $element['payment_errors'] = [
+      '#type' => 'markup',
+      '#markup' => '<div id="payment-errors"></div>',
+      '#weight' => -200,
+    ];
+
+    return $element;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitCreditCardForm(
+    array $element,
+    FormStateInterface $form_state
+  ) {
+    // The payment gateway plugin will process the submitted payment details.
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(
+    array $form,
+    FormStateInterface $form_state
+  ) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    // Add the bluesnap attribute to address form elements.
+    $form['billing_information']['address']['widget'][0]['address_line1']['#attributes']['data-bluesnap'] = 'address_line1';
+    $form['billing_information']['address']['widget'][0]['address_line2']['#attributes']['data-bluesnap'] = 'address_line2';
+    $form['billing_information']['address']['widget'][0]['locality']['#attributes']['data-bluesnap'] = 'address_city';
+    $form['billing_information']['address']['widget'][0]['postal_code']['#attributes']['data-bluesnap'] = 'address_zip';
+    $form['billing_information']['address']['widget'][0]['country_code']['#attributes']['data-bluesnap'] = 'address_country';
```

Comment:
@shabananavas Any reason for not doing the suggested changes? This is both not readable and violates coding standards.